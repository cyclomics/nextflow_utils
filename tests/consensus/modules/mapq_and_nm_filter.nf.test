nextflow_process {

    name "Test Process MapqAndNMFilter"
    script "consensus/modules/samtools.nf"
    process "MapqAndNMFilter"
    
    
    test("Filtering Mapq and NM") {

        when {
            process {
                """
                sample_id = "SOME_ID_1"
                file_id = "SOME_ID_2"
                bam_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.primary_mapped.bam")
                bai_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.primary_mapped.bam.bai")
                
                input[0] = Channel.of([sample_id, file_id, bam_file, bai_file])
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
            
            def expected_bai = "FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20.bam.bai"
            with(process.out[0]) {
                def result_bai = path(get(0)[3]).getFileName().toString()
                assert result_bai == expected_bai
            }
        }
    }
}
