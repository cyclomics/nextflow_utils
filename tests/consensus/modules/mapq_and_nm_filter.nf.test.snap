{
    "Filtering Mapq and NM": {
        "content": [
            {
                "0": [
                    [
                        "SOME_ID_1",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20.bam:md5,854ac1bb448ecb65b72e055bbcaf5ee6",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20.bam.bai:md5,86c925fb4e8860f4700b5582565d73e1"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-02-05T15:25:05.238335"
    }
}